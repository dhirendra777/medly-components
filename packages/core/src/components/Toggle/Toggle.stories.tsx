import { defaultTheme, ToggleTheme } from '@medly-components/theme';
import { boolean, select } from '@storybook/addon-knobs';
import React from 'react';
import { Toggle } from './Toggle';
import { Props } from './types';

const labelPosition: Array<Props['labelPosition']> = ['left', 'right', 'top', 'bottom'];
const size: Array<Props['size']> = ['XS', 'S', 'M', 'L', 'XL'];

export const ThemeInterface = (props: ToggleTheme): any => null;
ThemeInterface.defaultProps = {
    ...defaultTheme.checkbox
};

export const Basic = () => (
    <Toggle
        size={select('Size', size, 'S')}
        label="Orange"
        fullWidth={boolean('Full Width', false)}
        disabled={boolean('Disabled', false)}
        labelPosition={select('Label Position', labelPosition, 'right')}
    />
);
