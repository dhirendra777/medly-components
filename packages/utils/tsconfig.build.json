{
    "extends": "./tsconfig.json",
    "compilerOptions": {
        "outDir": "dist",
        "rootDir": "src",
        "baseUrl": "./",
        "noEmit": false,
        "declaration": true,
        "emitDeclarationOnly": true
    },
    "exclude": ["dist", "**/*.test.tsx", "src/test-utils.tsx", "**/*.test.ts", "**/*.stories.tsx"]
}
