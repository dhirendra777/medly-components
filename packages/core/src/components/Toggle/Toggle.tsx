import { WithStyle } from '@medly-components/utils';
import React, { SFC, useCallback } from 'react';
import FieldWithLabel from '../FieldWithLabel';
import * as Styled from './Toggle.styled';
import { Props } from './types';

export const Toggle: SFC<Props> & WithStyle = React.memo(
    React.forwardRef((props, ref) => {
        const { size, label, required, labelPosition, labelSize, labelWeight, labelColor, fullWidth, onChange, ...restProps } = props;

        const changeHandler = useCallback(
            (e: any) => {
                e.stopPropagation();
                onChange && onChange(e);
            },
            [onChange]
        );

        return (
            <FieldWithLabel id={`${label}-checkbox`} fieldWithMaxContent {...{ fullWidth, labelPosition }}>
                {label && (
                    <FieldWithLabel.Label
                        showPointer={!restProps.disabled}
                        {...{ required, labelPosition, labelSize, labelWeight, labelColor }}
                        htmlFor={label}
                    >
                        {label}
                    </FieldWithLabel.Label>
                )}
                <Styled.Wrapper size={size} disabled={restProps.disabled}>
                    {
                        // @ts-ignore
                        <Styled.Checkbox ref={ref} id={label} size={size} required={required} onChange={changeHandler} {...restProps} />
                    }
                    <Styled.Circle size={size} />
                </Styled.Wrapper>
            </FieldWithLabel>
        );
    })
);

Toggle.displayName = 'Toggle';
Toggle.Style = FieldWithLabel.Style;
Toggle.defaultProps = {
    label: '',
    required: false,
    labelPosition: 'right'
};
