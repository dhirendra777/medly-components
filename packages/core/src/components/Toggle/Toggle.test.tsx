import { TestUtils } from '@medly-components/utils';
import React from 'react';
import { Toggle } from './Toggle';

describe('Checkbox component', () => {
    it('should render correctly with all the default props', () => {
        const { container } = TestUtils.render(<Toggle name="gender" />);
        expect(container).toMatchSnapshot();
    });

    it('should render correctly with all the props given', () => {
        const { container } = TestUtils.render(<Toggle disabled fullWidth name="gender" label="Female" size="M" labelPosition="right" />);
        expect(container).toMatchSnapshot();
    });

    it('should call onChange handler on click on checkbox', () => {
        const mockOnChange = jest.fn();
        const { container } = TestUtils.render(
            <Toggle onChange={mockOnChange} disabled name="gender" label="Female" size="M" labelPosition="right" />
        );
        TestUtils.fireEvent.click(container.querySelector('input'));
        expect(mockOnChange).toBeCalled();
    });
});
