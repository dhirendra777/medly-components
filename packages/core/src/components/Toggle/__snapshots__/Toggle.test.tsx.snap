// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render correctly with all the default props 1`] = `
.c0 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-columns: max-content minmax(max-content,25%);
  grid-template-areas: 'field label' 'description random';
  grid-template-columns: auto;
  grid-template-areas: 'field' 'description';
}

.c0 .c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c1 {
  position: relative;
  width: calc(20px * 2);
  height: 20px;
  border-radius: 15px;
}

.c1:focus-within {
  box-shadow: 0 0 0 1pt #012040;
}

.c1 > * {
  width: 100%;
  height: 100%;
  border-radius: 15px;
}

.c4 {
  z-index: 1;
  position: absolute;
  background-color: #E0E0E0;
}

.c4::after {
  content: '';
  margin: 3px;
  display: block;
  border-radius: 50%;
  background: #ffffff;
  box-shadow: 1px 3px 3px 1px rgba(0,0,0,0.2);
  -webkit-transition: 0.2s;
  transition: 0.2s;
  width: calc(20px - 6px);
  height: calc(20px - 6px);
}

.c2 {
  opacity: 0;
  z-index: 2;
  margin: 0;
  position: absolute;
  cursor: pointer;
}

.c2:focus {
  outline: none;
}

.c2:disabled {
  cursor: not-allowed;
}

.c2:disabled + .c3 {
  background: #BDBDBD;
}

.c2:checked + .c3 {
  background: #012040;
}

.c2:checked + .c3::after {
  margin-left: calc(calc(20px * 2) - (20px) + 3px);
}

<div>
  <div
    class="c0"
    id="-checkbox"
  >
    
    <div
      class="c1"
    >
      <input
        class="c2"
        id=""
        name="gender"
        type="checkbox"
      />
      <div
        class="c3 c4"
      />
    </div>
  </div>
</div>
`;

exports[`Checkbox component should render correctly with all the props given 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 0px 0px 8px;
  cursor: not-allowed;
}

.c2 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  margin: 8px 0px;
  display: grid;
  grid-template-columns: max-content minmax(max-content,25%);
  grid-template-areas: 'field label' 'description random';
}

.c0 .c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c3 {
  position: relative;
  width: calc(28px * 2);
  height: 28px;
  border-radius: 15px;
}

.c3:focus-within {
  box-shadow: 0 0 0 1pt #012040;
}

.c3 > * {
  width: 100%;
  height: 100%;
  border-radius: 15px;
}

.c6 {
  z-index: 1;
  position: absolute;
  background-color: #E0E0E0;
}

.c6::after {
  content: '';
  margin: 3px;
  display: block;
  border-radius: 50%;
  background: #ffffff;
  box-shadow: 1px 3px 3px 1px rgba(0,0,0,0.2);
  -webkit-transition: 0.2s;
  transition: 0.2s;
  width: calc(28px - 6px);
  height: calc(28px - 6px);
}

.c7:disabled + .c5 {
  background: #BDBDBD;
}

.c7:checked + .c5 {
  background: #012040;
}

.c7:checked + .c5::after {
  margin-left: calc(calc(20px * 2) - (20px) + 3px);
}

.c4 {
  opacity: 0;
  z-index: 2;
  margin: 0;
  position: absolute;
  cursor: pointer;
}

.c4:focus {
  outline: none;
}

.c4:disabled {
  cursor: not-allowed;
}

.c4:disabled + .c5 {
  background: #BDBDBD;
}

.c4:checked + .c5 {
  background: #012040;
}

.c4:checked + .c5::after {
  margin-left: calc(calc(28px * 2) - (28px) + 3px);
}

<div>
  <div
    class="c0"
    id="Female-checkbox"
  >
    <label
      class="c1 c2"
      for="Female"
    >
      Female
    </label>
    <div
      class="c3"
      disabled=""
    >
      <input
        class="c4"
        disabled=""
        id="Female"
        name="gender"
        type="checkbox"
      />
      <div
        class="c5 c6"
      />
    </div>
  </div>
</div>
`;
